editorextensions/codebeamer-custom-ui/src/importmodal.ts

```diff
@@ -28,27 +29,79 @@ export class ImportModal extends Modal {
 
 	protected viewport = new Viewport(this.client);
 
-	imports: Map<number, {totalItems: number, finished: boolean}> = new Map();
+	imports: Map<number, { totalItems: number; finished: boolean }> = new Map();
+	cardBlocks: CardBlockProxy[] = [];
 
 	protected messageFromFrame(message: Message): void {
 		switch (message.action) {
-			case 'startImport': 
-				this.imports.set(message.payload.id, { totalItems: message.payload.totalItems, finished: false });
+			case 'startImport':
+				this.imports.set(message.payload.id, {
+					totalItems: message.payload.totalItems,
+					finished: false,
+				});
 				break;
 			case 'importItem':
 				this.createCard(message.payload.cardData);
 				this.imports.get(message.payload.importId)!.totalItems--;
-				if (this.imports.get(message.payload.importId)!.totalItems <= 1) {
+				if (
+					this.imports.get(message.payload.importId)!.totalItems <= 1
+				) {
 					this.imports.delete(message.payload.importId);
 					this.hide();
 				}
 				break;
+			case 'updateCard':
+				this.updateCard(
+					message.payload.cardData,
+					message.payload.cardBlockId
+				);
+				break;
+			case 'getCardBlocks':
+				const cardBlocks = this.viewport
+					.getCurrentPage()
+					?.allBlocks.filter(
+						(block) => block instanceof CardBlockProxy
+					) as CardBlockProxy[];
+
+				this.cardBlocks = cardBlocks;
+
+				const data = cardBlocks.map((cardBlock) => ({
+					cardBlock: cardBlock,
+					codebeamerItemId:
+						cardBlock.shapeData.get('codebeamerItemId'),
+				}));
+				this.sendMessage(JSON.stringify(data));
+				break;
```

Comment:
Do also outsource the logic here into its own function (like with `updateCard`). The `messageFromFrame` should only act as a gateway, which then calls the adequate functions, but doesn't contain much logic itself at all.  
Else, it will quickly (starting now) get unclear and hard to maintain.

You should easily be able to trim this down to two lines in here.

_(My original item import cases could potentially also use some refactoring in this regard)_